hardhat.config.ts

```diff
@@ -1,19 +1,51 @@
-import { HardhatUserConfig } from "hardhat/types";
-import "@nomiclabs/hardhat-waffle";
-import "hardhat-typechain";
+import { HardhatUserConfig } from 'hardhat/types'
+import '@nomiclabs/hardhat-vyper'
+import '@nomiclabs/hardhat-waffle'
+import 'hardhat-typechain'
+import './tasks'
```

Comment:
This actually creates a circular dependency.

When I've tried to build/compile locally for the first time, your tasks actually do require built/compiled artifacts, see here:

https://github.com/DePayFi/depay-ethereum-router/pull/13/files#diff-f218761d0b0ddddbb66a4d9939065c5bf47afc35f0be3ad37007be2cbfb0697aR6

Thats why it wasn't building as hardhat needs to be able to load the `hardhat.config.ts` file in order to compile artifacts.

I suggest to remove `import `./tasks`, create a dedicated `hardhat.config.tasks.ts` file and have that import `hardhat.config.ts` and we call hardhat with the task specific config file when we need it. So that we can still build/compile the contracts:

```
npx hardhat --config hardhat.config.tasks.ts --network ropsten curvefi:deploy
```